R/logit.R

```diff
@@ -50,51 +50,44 @@ mle_BFGS_logit <- function(design, outcome) {
   )
 }
 
-#' Response probability
-#'
-#' Calculate the response probability with the design and coefficient in the logit model.
-#'
-#' @details
-#'
-#' @param design matrix, the design matrix X for the predictors.
-#' @param beta vector, the linear coefficients.
-#'
-#' @return A probability vector for the response.
-#'
 outcome_est <- function(design, beta) {
   1 - 1 / (1 + exp(crossprod(t(design), beta)))
 }
 
-#' Update step for beta
-#'
-#' Calculate the update step for the linear coefficient beta using Newton's method.
-#'
-#' @details
-#'
-#' @param grad vector, the gradient of the logit likelihood.
-#' @param Hessian matrix, the Hessian matrix of the likelihood
-#'
-#' @return A moving step for the linear coefficient beta.
-#'
 update_step <- function(grad, Hessian) {
   step <- solve(Hessian, grad)
   return(step)
 }
 
-#' Hessian matrix calculation
-#'
-#' Calculate the Hessian matrix for the current beta.
-#'
-#' @details
-#'
-#' @param design matrix, the design matrix X for the predictors.
-#' @param beta vector, the linear coefficients.
-#'
-#' @return The Hessian matrix.
-#'
-make_Hessian <- function(design, beta) {
+make_W <- function(design, beta){
   p <- as.vector(1 / (1 + exp(crossprod(t(design), beta))))
-  Hessian <- -crossprod(design, p * design)
+  W <- p*(1-p)
+  return(W)
+}
+
+make_Hessian <- function(design, beta) {
+  W = make_W(design, beta)
+  Hessian <- -crossprod(design, W * design)
+}
```

Comment:
Try to be consistent with return style when possible. Here you use both return() as well as just the final evaluated line